### PR TITLE
Splits Demo from main package, adds tests, adds CI config

### PR DESCRIPTION
**Pull request by chrissrogers**
- Splits the demo into its own package structure at `./demo`
  - designed to function as a CodeSandbox template: https://codesandbox.io/s/github/recurly/react-recurly/tree/tests/demo
  - Utilizes `react-scripts`
- Adds `RecurlyProvider` and `useRecurly` tests
- Adds `.travis.yml`
- Updates `README`

### CODE REVIEW

File: test/support/helpers.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export function suppressConsoleErrors () {
+  beforeEach(() => {
+    jest.spyOn(console, 'error');
+    console.error.mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    console.error.mockRestore();
+  });
+}
```
Comment (chrissrogers) [inline test/support/helpers.js]: This method deactivates the virtual console error method. This is useful when we expect a test to produce errors.

File: test/use-recurly.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    });
+
+    describe('recurly.token', function () {
+      it('tokenizes the parent Elements instance', function () {
+        fixture(TestComponent)();
+
+        function TestComponent () {
+          const recurly = useRecurly();
+          const elements = React.useContext(RecurlyElementsContext).elements;
+          const underlyingRecurly = elements.recurly;
```
Comment (chrissrogers) [inline test/use-recurly.test.js]: The extraction of the underlying `Recurly` instance here relies on knowing that an `Elements` instance carries a reference to the `Recurly` that created it. While that's outside of the domain of this library, I think it's a reasonable access point in order to ensure that the `token` method is being called with the expected bound `Elements` instance.

File: test/provider.test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      render(
+        <RecurlyProvider publicKey="test-public-key">
+          <StubComponent />
+        </RecurlyProvider>
+      );
+
+      function StubComponent () {
+        const recurlyContext = useContext(RecurlyContext);
+        expect(recurlyContext.recurly).toBeInstanceOf(window.recurly.Recurly);
+        return '';
```
Comment (chrissrogers) [inline test/provider.test.js]: using the `RecurlyContext` in this way causes this test to leak an abstraction from the component under test, but it is the most effective way to extract the `Recurly` instance and introspect the effect of using `<RecurlyProvider />`

File: jest.setup.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import '@testing-library/jest-dom/extend-expect';
```
Comment (chrissrogers) [inline jest.setup.js]: This is actually not used in this PR, but I project it being needed for the `<CardElement />` and others, so added it.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,18 @@
-# react-recurly
+<img src="https://i.imgur.com/gGxJ8zx.png" align="right">
```
Comment (dbrudner) [inline README.md]: Needs more emojis 🚀

File: demo/src/card-element-demo.js [hunk 6]
```diff
@@ -1,12 +1,11 @@
 import React, { useState } from 'react';
 
-// in your app, these would be imported from 'react-recurly'
 import {
   CardElement,
   Elements,
   RecurlyProvider,
   useRecurly
-} from '../../lib/index';
+} from '@recurly/react-recurly';
```
Comment (dbrudner) [inline demo/src/card-element-demo.js]: 👍

File: test/provider.test.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+import { suppressConsoleErrors } from './support/helpers';
+
+import { RecurlyProvider } from '../lib';
+import { RecurlyContext } from '../lib/provider';
+
+describe('<RecurlyProvider />', function () {
+  describe('without having included recurly.js', function () {
+    const storedRecurly = window.recurly;
+
+    suppressConsoleErrors();
```
Comment (dbrudner) [inline test/provider.test.js]: A comment explaining why this is here could be helpful

File: test/provider.test.js (hunk 3 above)
Comment (dbrudner) [reply test/provider.test.js]: I'm not sure what you mean by "leak an abstraction", but this looks fine

File: test/support/helpers.js (hunk 1 above)
Comment (dbrudner) [reply test/support/helpers.js]: It might be useful to add this as a comment in the code as well

Comment (chrissrogers) [reply test/support/helpers.js]: Good call -- done!

**Pull request merged**